src/Attributes.jl

```diff
@@ -113,7 +113,7 @@ macro attributes(expr)
    error("attributes can only be attached to mutable structs")
 end
 
-_is_attribute_storing_type(::Type{T}) where T = Base.issingletontype(T) || isstructtype(T) && ismutable(T) && hasfield(T, :__attrs)
+_is_attribute_storing_type(::Type{T}) where T = Base.issingletontype(T) || isstructtype(T) && AbstractAlgebra.ismutabletype(T) && hasfield(T, :__attrs)
```

Comment:
It is probably obvious, but why not just `ismutabletype(T)`?

Reply:
See https://github.com/Nemocas/AbstractAlgebra.jl/pull/1669#issuecomment-2060877809

Reply:
I copy&pasted it from the macro code, but I guess the macro code does it because it is executed in a "global" context. Here in this place I guess I could change it to `ismutabletype` which should pick up the definition we make earlier in the file